compat/src/suspense.js

```diff
@@ -30,24 +24,20 @@ options._catchError = function(error, newVNode, oldVNode) {
 	oldCatchError(error, newVNode, oldVNode);
 };
 
-function detachDom(children) {
-	for (let i = 0; i < children.length; i++) {
-		let child = children[i];
-		if (child != null) {
-			if (typeof child.type !== 'function' && child._dom) {
-				removeNode(child._dom);
-			} else if (child._children) {
-				detachDom(child._children);
-			}
+function detachDom(vnode) {
+	if (vnode) {
+		if (typeof vnode.type !== 'function' && vnode._dom) {
+			removeNode(vnode._dom);
+		} else if (vnode._children) {
+			vnode._children.some(detachDom);
 		}
 	}
 }
 
 // having custom inheritance instead of a class here saves a lot of bytes
 export function Suspense(props) {
 	// we do not call super here to golf some bytes...
-	this._suspensions = [];
-	this._fallback = props.fallback;
```

Comment:
was there some reason for storing this on the class?

Reply:
I'm not quite sure, but it might have had something to do how Suspense was implemented before this PR.

Reply:
https://github.com/preactjs/preact/commit/17f65e4dfd209d643117b0fab507c43eacdb4b96

Yeah probably some "legacy things"™

Reply:
The fallback is later cloned using `cloneElement` to not re-use the vnode as this is not supported in our current code base. See line 87 prior to this PR

Reply:
I guess the cloning is no longer needed as the fallback is now rendered as a child of a Fragment that is then cloned in coerceToVNode anyways if it was already used.

Reply:
Yeah I suppose so.

Reply:
I think the problem was that the old implementation was manually calling `unmount`. So we had to make sure that the VNode that was called with `unmount` was the most up to date VNode. If we re-used a VNode (e.g. if a component re-suspends), `toChildArray` notices that this VNode has been used before (e.g. it has a `_component` property) and clones it. In the old implementation this was a problem because it mean that the reference to the `props.fallback` VNode was no longer in sync with the VNode that the diff was using.

To fix this, we cloned the `fallback` before going into the diff to skip the `toChildArray` clone, guaranteeing our reference always matched the one the diff was using.

Since this PR changes suspense to no longer manually call `unmount` we longer need to worry about VNode references, since we rely on the natural thing that `diff` does!

(This is from my memory of the previous PR, didn't re-verify it so I hope my memory isn't too off base lol)